src/crimson/osd/pg_recovery.cc

```diff
@@ -533,10 +533,10 @@ void PGRecovery::enqueue_push(
 {
   logger().info("{}: obj={} v={} peers={}", __func__, obj, v, peers);
   auto &peering_state = pg->get_peering_state();
-  peering_state.prepare_backfill_for_missing(obj, v, peers);
   auto [recovering, added] = pg->get_recovery_backend()->add_recovering(obj);
   if (!added)
     return;
+  peering_state.prepare_backfill_for_missing(obj, v, peers);
```

Comment:
nit: This commit is not related to the previous commit of this PR, right?

Reply:
Um, I think it's related, because an object may be enqueued for push repeatly by BackfillState with the previous commit, so we need to prevent it from making harmful changes.